// @dicebear/shapes/lib/utils/getComponents.js
/**
 * Do not change this file manually! This file was generated with the "Dicebear Exporter"-Plugin for Figma.
 *
 * Plugin: https://www.figma.com/community/plugin/1005765655729342787
 * File: https://www.figma.com/file/bX8ZT8jK2fo5Uy8G6j2Qic
 */
import { pickComponent } from './pickComponent.js';
export function getComponents({ prng, options }) {
    var _a, _b, _c, _d, _e, _f, _g, _h, _j;
    const shape1Component = pickComponent({
        prng,
        group: 'shape1',
        values: options.shape1,
        width: 100,
        height: 100,
        rotation: ((_a = options.shape1Rotation) === null || _a === void 0 ? void 0 : _a.length) ? options.shape1Rotation : [0],
        offsetX: ((_b = options.shape1OffsetX) === null || _b === void 0 ? void 0 : _b.length) ? options.shape1OffsetX : [0],
        offsetY: ((_c = options.shape1OffsetY) === null || _c === void 0 ? void 0 : _c.length) ? options.shape1OffsetY : [0],
    });
    const shape2Component = pickComponent({
        prng,
        group: 'shape2',
        values: options.shape2,
        width: 100,
        height: 100,
        rotation: ((_d = options.shape2Rotation) === null || _d === void 0 ? void 0 : _d.length) ? options.shape2Rotation : [0],
        offsetX: ((_e = options.shape2OffsetX) === null || _e === void 0 ? void 0 : _e.length) ? options.shape2OffsetX : [0],
        offsetY: ((_f = options.shape2OffsetY) === null || _f === void 0 ? void 0 : _f.length) ? options.shape2OffsetY : [0],
    });
    const shape3Component = pickComponent({
        prng,
        group: 'shape3',
        values: options.shape3,
        width: 100,
        height: 100,
        rotation: ((_g = options.shape3Rotation) === null || _g === void 0 ? void 0 : _g.length) ? options.shape3Rotation : [0],
        offsetX: ((_h = options.shape3OffsetX) === null || _h === void 0 ? void 0 : _h.length) ? options.shape3OffsetX : [0],
        offsetY: ((_j = options.shape3OffsetY) === null || _j === void 0 ? void 0 : _j.length) ? options.shape3OffsetY : [0],
    });
    return {
        'shape1': shape1Component,
        'shape2': shape2Component,
        'shape3': shape3Component,
    };
}
;
